pimd/pim_cmd_common.c

```diff
@@ -653,3 +663,880 @@ int pim_process_no_rp_plist_cmd(struct vty *vty, const char *rp_str,
 
 	return nb_cli_apply_changes(vty, NULL);
 }
+
+bool pim_sgaddr_match(pim_sgaddr item, pim_sgaddr match)
+{
+	return (pim_addr_is_any(match.grp) ||
+		!pim_addr_cmp(match.grp, item.grp)) &&
+	       (pim_addr_is_any(match.src) ||
+		!pim_addr_cmp(match.src, item.src));
+}
+
+void json_object_pim_upstream_add(json_object *json, struct pim_upstream *up)
+{
+	if (up->flags & PIM_UPSTREAM_FLAG_MASK_DR_JOIN_DESIRED)
+		json_object_boolean_true_add(json, "drJoinDesired");
```

Comment:
I would say, let's add all those fields despite the flag existing or not. Just to easier the automation for machines.